azwi: Try to run create app results in segmentation violation in v0.7.0

**Describe the bug**

Running `azwi serviceaccount create phase app --aad-application-name "${APPLICATION_NAME}"` results in a segmentation violation

**Steps To Reproduce**

Install azwi from [github release](https://github.com/Azure/azure-workload-identity/releases/download/v0.7.0/azwi-v0.7.0-linux-amd64.tar.gz)

Follow [step 4](https://azure.github.io/azure-workload-identity/docs/quick-start.html#4-create-an-aad-application-and-grant-permissions-to-access-the-secret) of the quick start

```
joelp@host:~$ azwi serviceaccount create phase app --aad-application-name "${APPLICATION_NAME}"
INFO[0000] No subscription provided, using selected subscription from Azure CLI: xxxx-xxxx-xxxx-xxxx
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4122770]

goroutine 1 [running]:
github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/create.(*aadApplicationPhase).run(0xc000707b68, {0x60eb9f0, 0xc000146048}, {0x5884640, 0xc000530820})
        /Users/ernestwong/go/src/github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/create/aadapplication.go:68 +0x190
github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/workflow.(*runner).Run(0xc00049d620, {0x5884640, 0xc000530820})
        /Users/ernestwong/go/src/github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/workflow/runner.go:124 +0x163
github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/workflow.(*runner).BindToCommand.func1(0xc0004c4f00, {0x5a3910a, 0x2, 0x2})
        /Users/ernestwong/go/src/github.com/Azure/azure-workload-identity/pkg/cmd/serviceaccount/phases/workflow/runner.go:90 +0xd0
github.com/spf13/cobra.(*Command).execute(0xc0004c4f00, {0xc0004309e0, 0x2, 0x2})
        /Users/ernestwong/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004c4280)
        /Users/ernestwong/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/ernestwong/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
        /Users/ernestwong/go/src/github.com/Azure/azure-workload-identity/cmd/azwi/main.go:15 +0x6f
```

**Expected behavior**

Command works properly

**Environment**

- Kubernetes version (use `kubectl version`): 
  ```
  Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
  Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"802eff1fe87ad2dd737ebbe891f30500b88beb00", GitTreeState:"clean", BuildDate:"2021-11-15T08:35:41Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
  ```
- Cloud provider or hardware configuration: Azure
- OS (e.g: `cat /etc/os-release`):
  ```
  NAME="Ubuntu"
  VERSION="20.04.3 LTS (Focal Fossa)"
  ID=ubuntu
  ID_LIKE=debian
  PRETTY_NAME="Ubuntu 20.04.3 LTS"
  VERSION_ID="20.04"
  HOME_URL="https://www.ubuntu.com/"
  SUPPORT_URL="https://help.ubuntu.com/"
  BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
  PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
  VERSION_CODENAME=focal
  UBUNTU_CODENAME=focal
  ```
- Kernel (e.g. `uname -a`): `Linux APEAC2VDD0009 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

--- a/pkg/cloud/graph.go
+++ b/pkg/cloud/graph.go
@@ -28,7 +28,18 @@ func (c *AzureClient) CreateServicePrincipal(ctx context.Context, appID string, 
 	spPostOptions.Body.SetTags(tags)
 
 	log.Debugf("Creating service principal for application with id=%s", appID)
-	return c.graphServiceClient.ServicePrincipals().Post(spPostOptions)
+	sp, err := c.graphServiceClient.ServicePrincipals().Post(spPostOptions)
+	if err != nil {
+		return nil, err
+	}
+	graphErr, err := GetGraphError(sp.GetAdditionalData())
+	if err != nil {
+		return nil, err
+	}
+	if graphErr != nil {
+		return nil, *graphErr
+	}
+	return sp, nil
 }
 
 // CreateApplication creates an application.
@@ -39,7 +50,18 @@ func (c *AzureClient) CreateApplication(ctx context.Context, displayName string)
 	appPostOptions.Body.SetDisplayName(to.StringPtr(displayName))
 
 	log.Debugf("Creating application with display name=%s", displayName)
-	return c.graphServiceClient.Applications().Post(appPostOptions)
+	app, err := c.graphServiceClient.Applications().Post(appPostOptions)
+	if err != nil {
+		return nil, err
+	}
+	graphErr, err := GetGraphError(app.GetAdditionalData())
+	if err != nil {
+		return nil, err
+	}
+	if graphErr != nil {
+		return nil, *graphErr
+	}
+	return app, nil
 }
 
 // GetServicePrincipal gets a service principal by its display name.
@@ -55,6 +77,13 @@ func (c *AzureClient) GetServicePrincipal(ctx context.Context, displayName strin
 	resp, err := c.graphServiceClient.ServicePrincipals().Get(spGetOptions)
 	if err != nil {
 		return nil, err
+	}
+	graphErr, err := GetGraphError(resp.GetAdditionalData())
+	if err != nil {
+		return nil, err
+	}
+	if graphErr != nil {
+		return nil, *graphErr
 	}
 	if len(resp.GetValue()) == 0 {
 		return nil, errors.Errorf("service principal %s not found", displayName)
@@ -75,6 +104,13 @@ func (c *AzureClient) GetApplication(ctx context.Context, displayName string) (*
 	resp, err := c.graphServiceClient.Applications().Get(appGetOptions)
 	if err != nil {
 		return nil, err
+	}
+	graphErr, err := GetGraphError(resp.GetAdditionalData())
+	if err != nil {
+		return nil, err
+	}
+	if graphErr != nil {
+		return nil, *graphErr
 	}
 	if len(resp.GetValue()) == 0 {
 		return nil, errors.Errorf("application with display name '%s' not found", displayName)

--- a/pkg/cmd/serviceaccount/root.go
+++ b/pkg/cmd/serviceaccount/root.go
@@ -15,6 +15,10 @@ func NewServiceAccountCmd() *cobra.Command {
 		Long:    "Manage the workload identity",
 		Aliases: []string{"sa"},
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// run root command pre-run to registry the debug flag
+			if cmd.Root() != nil && cmd.Root().PersistentPreRun != nil {
+				cmd.Root().PersistentPreRun(cmd.Root(), args)
+			}
 			return authProvider.Validate()
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {

--- a/pkg/cmd/serviceaccount/root.go
+++ b/pkg/cmd/serviceaccount/root.go
@@ -15,7 +15,7 @@ func NewServiceAccountCmd() *cobra.Command {
 		Long:    "Manage the workload identity",
 		Aliases: []string{"sa"},
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			// run root command pre-run to registry the debug flag
+			// run root command pre-run to register the debug flag
 			if cmd.Root() != nil && cmd.Root().PersistentPreRun != nil {
 				cmd.Root().PersistentPreRun(cmd.Root(), args)
 			}

